core/src/main/java/org/jruby/javasupport/proxy/JavaProxyClassFactory.java

```diff
@@ -155,15 +160,43 @@ public JavaProxyClass newProxyClass(final Ruby runtime, ClassDefiningClassLoader
         validateArgs(runtime, targetClassName, superClass);
 
         Type selfType = Type.getType('L' + toInternalClassName(targetClassName) + ';');
-        Map<MethodKey, MethodData> methods = collectMethods(superClass, interfaces, names);
+        Map<MethodKey, MethodData> methods = collectMethods(superClass, interfaces, names, extraMethods, clazz.getMethodAnnotations());
 
-        return generate(loader, targetClassName, superClass, interfaces, methods, selfType);
+        return generate(loader, targetClassName, superClass, interfaces, methods, selfType, clazz);
     }
 
     private JavaProxyClass generate(ClassDefiningClassLoader loader, String targetClassName,
                                     Class superClass, Class[] interfaces,
-                                    Map<MethodKey, MethodData> methods, Type selfType) {
+                                    Map<MethodKey, MethodData> methods, Type selfType, RubyClass rclass) {
         ClassWriter cw = beginProxyClass(targetClassName, superClass, interfaces, loader);
+        
+        Map<String, Map<Class, Map<String, Object>>> fieldannos;
+        Map<String, Class> fields;
+        if (rclass != null)
+        {
+        	fieldannos = rclass.getFieldAnnotations();
+        	fields = rclass.getFieldSignatures();
+        	
+        }
+        else 
+        {
```

Comment:
Style issues here and other places... close brace and open brace should be on same line as else, if's open brace should be on same line, etc.